my_pyscf/grad/lasscf.py

```diff
@@ -0,0 +1,258 @@
+        return mcscf_grad
+
+    logger.info(mcscf_grad, 'Create scanner for %s', mcscf_grad.__class__)
+    name = mcscf_grad.__class__.__name__ + CASSCF_GradScanner.__name_mixin__
+    return lib.set_class(CASSCF_GradScanner(mcscf_grad),
+                         (CASSCF_GradScanner, mcscf_grad.__class__), name)
+
+class CASSCF_GradScanner(lib.GradScanner):
+    def __init__(self, g):
+        lib.GradScanner.__init__(self, g)
+
+    def __call__(self, mol_or_geom, **kwargs):
+        if isinstance(mol_or_geom, gto.MoleBase):
+            assert mol_or_geom.__class__ == gto.Mole
+            mol = mol_or_geom
+        else:
+            mol = self.mol.set_geom_(mol_or_geom, inplace=False)
+        self.reset(mol)
+
+        mc_scanner = self.base
+        e_tot = mc_scanner(mol)
+        if isinstance(mc_scanner, StateAverageMCSCFSolver):
+            e_tot = mc_scanner.e_average
+
+        de = self.kernel(**kwargs)
+        return e_tot, de
+
+
+class Gradients(casci_grad.Gradients):
+    '''Non-relativistic restricted Hartree-Fock gradients'''
+    print ("SV entering Gradients in LASSCF")
+    grad_elec = grad_elec
+
+    def kernel(self, mo_coeff=None, ci=None, atmlst=None, verbose=None):
+        print ("SV entering kernel")
+        log = logger.new_logger(self, verbose)
+        if atmlst is None:
+            atmlst = self.atmlst
+        else:
+            self.atmlst = atmlst
+
+        if self.verbose >= logger.WARN:
+            self.check_sanity()
+        if self.verbose >= logger.INFO:
+            self.dump_flags()
+
+        de = self.grad_elec(mo_coeff, ci, atmlst, log)
+        self.de = de = de + self.grad_nuc(atmlst=atmlst)
+        if self.mol.symmetry:
+            self.de = self.symmetrize(self.de, atmlst)
+        self._finalize()
+        return self.de
+    '''
+    #SV Added hcore_generator and grad_nuc
+    # Initialize hcore_deriv with the underlying SCF object because some
+    # extensions (e.g. x2c, QM/MM, solvent) modifies the SCF object only.
+    def hcore_generator(self, mol=None):
+        mf_grad = self.base._scf.nuc_grad_method()
+        return mf_grad.hcore_generator(mol)
+
+    # Calling the underlying SCF nuclear gradients because it may be modified
+    # by external modules (e.g. QM/MM, solvent)
+    def grad_nuc(self, mol=None, atmlst=None):
+        mf_grad = self.base._scf.nuc_grad_method()
+        return mf_grad.grad_nuc(mol, atmlst)
+
+    #SV addition ended
+    '''
+    def _finalize(self):
+        if self.verbose >= logger.NOTE:
+            logger.note(self, '--------------- %s gradients ---------------',
+                        self.base.__class__.__name__)
+            self._write(self.mol, self.de, self.atmlst)
+            logger.note(self, '----------------------------------------------')
+
+    as_scanner = as_scanner
+
+    to_gpu = lib.to_gpu
+
+Grad = Gradients
+
+from pyscf import mcscf
+mcscf.mc1step.CASSCF.Gradients = lib.class_as_method(Gradients)
```

Comment:
Delete this! It messes up CASSCF gradients!

Reply:
done